### PR TITLE
kube-inject: Add support to Openshift DeploymentConfigs

### PR DESCRIPTION
**Pull request by jwendell**
Addresses #1912 in istioctl kube-inject.

### CODE REVIEW

File: pilot/pkg/kube/inject/initializer.go [hunk 1]
```diff
@@ -40,6 +41,7 @@ var (
 		{v1.SchemeGroupVersion, &v1.ReplicationController{}, "replicationcontrollers", "/api"},
 
 		{v1beta1.SchemeGroupVersion, &v1beta1.Deployment{}, "deployments", "/apis"},
+		{v1.SchemeGroupVersion, &openshiftv1.DeploymentConfig{}, "deploymentconfigs", "/apis"},
```
Comment (ayj) [inline pilot/pkg/kube/inject/initializer.go]: i think this should be use `openshiftv1.LegacySchemeGroupVersion` instead.In addition,  we also probably want the newer `openshiftv1.SchemeGroupversion` as in the list of supported kinds.

Comment (jwendell) [reply pilot/pkg/kube/inject/initializer.go]: Makes total sense. Just updated the code.

File: pilot/pkg/kube/inject/initializer.go [hunk 2]
```diff
@@ -49,6 +50,10 @@ var (
 		{appsv1beta1.SchemeGroupVersion, &appsv1beta1.StatefulSet{}, "statefulsets", "/apis"},
 
 		{v1.SchemeGroupVersion, &v1.List{}, "lists", "/apis"},
+
+		// Openshift
```
Comment (ayj) [inline pilot/pkg/kube/inject/initializer.go]: The comment is probably not needed given the package naming :)

**Comment by andraxylia** [comment]
@jwendell please resolve conflicts, it will re-test.

**Pull request merged**